### PR TITLE
Riffsy: Fixes #2473

### LINKED ISSUE #2473: Riffsy: Restrict character matching to Emoji only
Gifs that match Emoji zeroclick doesn't just match Emoji, but other high unicode characters as well. It should restrict it to only match the appropriate range of characters.

------
IA Page:  http://duck.co/ia/view/riffsy
Maintainer: @bryanhart

### PR DESCRIPTION
**Pull request by swapagarwal**
Verify the unicode character matched is an emoji using [official Unicode standards](http://www.unicode.org/Public/UNIDATA/EmojiSources.txt).
IA Page: http://duck.co/ia/view/riffsy
Maintainer: @bryanhart

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Spice/Riffsy.pm [hunk 1]
```diff
@@ -10,10 +10,13 @@ spice proxy_cache_valid => "200 1d";
 spice to => 'http://api1.riffsy.com/v1/search?tag=$1&key={{ENV{DDG_SPICE_RIFFSY_APIKEY}}}&safesearch=strict&searchtype=ddg&limit=50';
 spice wrap_jsonp_callback => 1;
 
-triggers query_lc => qr/^\X$|^(\X+|[\w\s]+?) gifs?$/;
+triggers query_lc => qr/^(?<match>\X)$|^(\X+|[\w\s]+?) gifs?$/;
 
 handle query_lc => sub {
     return if !$_ || $_ =~ /^[a-zA-Z0-9]$/;
+    my @emojis = map { chomp $_; $_ } share('emoji_sources.txt')->slurp;
```
Comment (moollaza) [inline lib/DDG/Spice/Riffsy.pm]: Let's move this outside the `handle` so we only build this regex once and store it in memory for the `handle` function to use when needed

Comment (swapagarwal) [inline lib/DDG/Spice/Riffsy.pm]: Sure, makes perfect sense.

**Comment by moollaza** [comment]
@swapagarwal so sorry for the delay. This is excellent! Very nicely done 💯 👍

**Comment by swapagarwal** [comment]
Thanks @moollaza! :smile:

**Pull request merged**